stereo_image_proc/src/nodelets/point_cloud2.cpp

```diff
@@ -231,18 +217,10 @@ void PointCloud2Nodelet::imageCb(const ImageConstPtr& l_image_msg,
                                   l_image_msg->step);
     for (int v = 0; v < mat.rows; ++v)
     {
-      for (int u = 0; u < mat.cols; ++u, offset += STEP)
+      for (int u = 0; u < mat.cols; ++u, ++iter_r, ++iter_g, ++iter_b)
       {
-        if (isValidPoint(mat(v,u)))
-        {
-          uint8_t g = color(v,u);
-          int32_t rgb = (g << 16) | (g << 8) | g;
-          memcpy (&points_msg->data[offset + 12], &rgb, sizeof (int32_t));
-        }
-        else
-        {
-          memcpy (&points_msg->data[offset + 12], &bad_point, sizeof (float));
-        }
+        uint8_t g = color(v,u);
+        *iter_r = *iter_g = *iter_b = g;
```

Comment:
that seems to change the behavior: before, when the point was not valid (e.g. MISSING_Z which is 10000), it would be set to infinity. Now, it is se to 10000